resources/js/entrypoints/beatmap-discussions.tsx

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import Main from 'beatmap-discussions/main';
+import core from 'osu-core-singleton';
+import React from 'react';
+import { parseJson } from 'utils/json';
+
+core.reactTurbolinks.register('beatmap-discussions', (container: HTMLElement) => (
+  <Main
+    container={container}
+    initial={parseJson('json-beatmapset-discussion')}
```

Comment:
this will just be an unused parsing if there's an existing state. Move the parsing in the component itself maybe?